Warning: enum out of range

I am getting the following warning when compiling my bindings against libgit2 0.27.0 with `-pedantic`:

```
./libgit2-0.27.0/include/git2/diff.h:214:30: warning: ISO C restricts enumerator values to range of 'int' [-Wpedantic]
```

Unfortunately my repository maintainers insist that the code can be compiled without any `-pedantic` warnings. Here is the offending line:

https://github.com/libgit2/libgit2/blob/d11c4a1a464f10c69d5cc58824e980ea5045d439/include/git2/diff.h#L210-L215

--- a/include/git2/diff.h
+++ b/include/git2/diff.h
@@ -171,6 +171,12 @@ typedef enum {
 	 * Options controlling how output will be generated
 	 */
 
+	/** Use a heuristic that takes indentation and whitespace into account
+	 * which generally can produce better diffs when dealing with ambiguous
+	 * diff hunks.
+	 */
+	GIT_DIFF_INDENT_HEURISTIC = (1u << 18),
+
 	/** Treat all files as text, disabling binary attributes & detection */
 	GIT_DIFF_FORCE_TEXT = (1u << 20),
 	/** Treat all files as binary, disabling text diffs */
@@ -206,12 +212,6 @@ typedef enum {
 	 *  can apply given diff information to binary files.
 	 */
 	GIT_DIFF_SHOW_BINARY = (1u << 30),
-
-	/** Use a heuristic that takes indentation and whitespace into account
-	 * which generally can produce better diffs when dealing with ambiguous
-	 * diff hunks.
-	 */
-	GIT_DIFF_INDENT_HEURISTIC = (1u << 31),
 } git_diff_option_t;
 
 /**

